src/Illuminate/Foundation/Testing/TestCase.php

```diff
@@ -262,4 +269,116 @@ protected function callBeforeApplicationDestroyedCallbacks()
             }
         }
     }
+
+    /**
+     * This method is called when a test method did not execute successfully.
+     *
+     * @param  \Throwable  $exception
+     * @return void
+     */
+    protected function onNotSuccessfulTest(Throwable $exception): void
+    {
+        if (! $exception instanceof ExpectationFailedException || is_null($this->latestResponse)) {
+            parent::onNotSuccessfulTest($exception);
+        }
+
+        if ($lastException = $this->latestResponse->exceptions->last()) {
+            parent::onNotSuccessfulTest($this->appendExceptionToException($lastException, $exception));
+
+            return;
+        }
+
+        if ($this->latestResponse->baseResponse instanceof RedirectResponse) {
+            $session = $this->latestResponse->baseResponse->getSession();
+
+            if (! is_null($session) && $session->has('errors')) {
+                parent::onNotSuccessfulTest($this->appendErrorsToException($session->get('errors')->all(), $exception));
+
+                return;
+            }
+        }
+
+        if ($this->latestResponse->baseResponse->headers->get('Content-Type') === 'application/json') {
+            $testJson = new AssertableJsonString($this->latestResponse->getContent());
+
+            if (isset($testJson['errors'])) {
```

Comment:
@jessarcher Should we check that the status code is also 422?

Reply:
The existing behaviour doesn't check this and I think that's okay. We don't mention anything about validation in the messaging, so I see it as an added bonus that this will work for users returning their own error payload for scenarios other than validation.